src/naive_bayes/mod.rs

```diff
@@ -0,0 +1,51 @@
+use crate::error::Failed;
+use crate::linalg::BaseVector;
+use crate::linalg::Matrix;
+use crate::math::num::RealNumber;
+pub trait NBDistribution {
+    // Fit distribution to some continuous or discrete data
+    fn fit<T: RealNumber, M: Matrix<T>>(x: &M, y: &M::RowVector) -> Self;
+
+    // Prior of class k
+    fn prior<T: RealNumber>(&self, k: T) -> T;
+
+    // Conditional probability of feature j give class k
+    fn conditional_probability<T: RealNumber, M: Matrix<T>>(&self, k: T, j: &M::RowVector) -> T;
+
+    // Possible classes of the distribution
+    fn classes<T: RealNumber, M: Matrix<T>>(&self) -> M::RowVector;
```

Comment:
I think this function should return simple Vec<T> since it represents an unordered list of classes that doesn't benefit from Vector methods like dot product, norm, e.t.c.